llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -523,6 +633,29 @@ int InterBlockScheduling::getSafetyMargin(MachineBasicBlock *Loop,
   return SafetyMargin;
 }
 
+void InterBlockScheduling::emitBundles(
+    const std::vector<MachineBundle> &Bundles, MachineBasicBlock *BB,
+    MachineBasicBlock::iterator Before, bool Move) const {
+  for (auto &Bundle : Bundles) {
+    if (Bundle.empty()) {
+      TII->insertNoop(*BB, Before);
+      continue;
+    }
+    bool InBundle = false;
+    for (auto *MI : Bundle.getInstrs()) {
+      if (Move) {
+        BB->remove_instr(MI);
+      }
+      BB->insert(Before, MI);
```

Comment:
Is there an equivalent `insert_instr`? Or is it on purpose so that we insert inside an existing bundle?

Reply:
I can never get my head around these issues. My assumption is that at this point we have no bundles, and we insert instructions before other instructions, leaving BundledWithPred markers to really create the bundles afterwards.